src/NuGet.Core/NuGet.Packaging/Core/ExtractPackageFileDelegate.cs

```diff
@@ -11,6 +11,7 @@ namespace NuGet.Packaging.Core
     /// <param name="sourceFile">The path of the file in the package.</param>
     /// <param name="targetPath">The path to write to.</param>
     /// <param name="fileStream">The file <see cref="Stream"/>.</param>
+    /// <param name="size">The size of the incoming file.</param>
     /// <returns>The file name if the file was written; otherwise <c>null</c>.</returns>
-    public delegate string ExtractPackageFileDelegate(string sourceFile, string targetPath, Stream fileStream);
+    public delegate string ExtractPackageFileDelegate(string sourceFile, string targetPath, Stream fileStream, long? size);
```

Comment:
Unfortunately this also is a breaking change, so another delegate is needed, and all methods using it need an overload.

I imagine we won't be able to make the old overloads simply call the new overload, because methods using the old delegate won't have any way to get the new delegate. This suggests, we need to keep codepaths that never use mmap. We can mark them as obsolete, assuming none of our production codepaths use it.

I wish more thought was made in the past about what should be public and what should be internal :(

Reply:
@zkat might be a good idea to rebase to get the public API analyzer in. 

Otherwise dev will break when this PR is merged.